src/applications/bmqtool/m_bmqtool_poster.cpp

```diff
@@ -0,0 +1,248 @@
+                        // timestamp:
+                        int nbMsgPerSec = d_parameters_p->eventSize() *
+                                          d_parameters_p->postRate() * 1000 /
+                                          d_parameters_p->postInterval();
+                        d_msgUntilNextTimestamp = nbMsgPerSec *
+                                                  k_LATENCY_INTERVAL_MS / 1000;
+                    }
+
+                    bdlbb::BlobBuffer buffer;
+                    d_timeBufferFactory_p->allocate(&buffer);
+                    buffer.setSize(sizeof(bdlb::BigEndianInt64));
+                    bsl::memcpy(buffer.buffer().get(),
+                                &timeNs,
+                                sizeof(timeNs));
+                    d_blob.swapBufferRaw(0, &buffer);
+                }
+                msg.setDataRef(&d_blob);
+
+                length = d_blob.length();
+            }
+
+            if (d_properties.numProperties()) {
+                msg.setPropertiesRef(&d_properties);
+            }
+            bmqt::EventBuilderResult::Enum rc = eventBuilder.packMessage(
+                d_queueId);
+            if (rc != 0) {
+                BALL_LOG_ERROR << "Failed to pack message [rc: " << rc << "]";
+                continue;  // CONTINUE
+            }
+            d_statContext_p->adjustValue(k_STAT_MSG, length);
+        }
+
+        // Now publish the event
+        const bmqa::MessageEvent& messageEvent = eventBuilder.messageEvent();
+
+        // Write PUTs to log file before posting
+        if (d_fileLogger && d_fileLogger->isOpen()) {
+            bmqa::MessageIterator it = messageEvent.messageIterator();
+            while (it.nextMessage()) {
+                const bmqa::Message& message = it.message();
+                d_fileLogger->writePutMessage(message);
+            }
+        }
+
+        int rc = d_session_p->post(messageEvent);
+
+        if (rc != 0) {
+            BALL_LOG_ERROR << "Failed to post: " << bmqt::PostResult::Enum(rc)
+                           << " (" << rc << ")";
+            continue;  // CONTINUE
+        }
+
+        const bsl::shared_ptr<bmqimp::Event>& eventImpl =
+            reinterpret_cast<const bsl::shared_ptr<bmqimp::Event>&>(
+                messageEvent);
+        d_statContext_p->adjustValue(k_STAT_EVT,
+                                     eventImpl->rawEvent().blob()->length());
+
+        if (d_parameters_p->eventsCount() > 0) {
+            --d_remainingEvents;
+        }
+    }
+}
+
+Poster::Poster(FileLogger*         fileLogger,
+               mwcst::StatContext* statContext,
+               bslma::Allocator*   allocator)
+: d_allocator_p(allocator)
+, d_bufferFactory(4096, d_allocator_p)
+, d_timeBufferFactory(sizeof(bdlb::BigEndianInt64), d_allocator_p)
+, d_statContext(statContext)
+, d_fileLogger(fileLogger)
+{
+    BSLS_ASSERT_SAFE(fileLogger);
+    BSLS_ASSERT_SAFE(statContext);
+    BSLS_ASSERT_SAFE(allocator);
```

Comment:
We don't want to support default arg=0 allocator here, right?
`bslma::Allocator* alloc = bslma::Default::allocator(allocator);`